docs/adr/001_error_handling.md

```diff
@@ -0,0 +1,107 @@
+# Error handling patterns in public API interfaces
+
+## Date
+17 Feb 2025 
+
+## Accepted Option
+
+**Option 3** 
```

Comment:
I suggest to make it super easy for future reader to know what is the design guidance here, without having to scan through rest of doc. i.e Let's put the followed design right here, and a why that was chosen just below it.

Everything else can be moved to bottom of the doc. https://github.com/open-telemetry/opentelemetry-go/blob/main/sdk/log/DESIGN.md#rejected-alternatives has something like this.